sui_programmability/examples/nft_standard/README.md

```diff
@@ -0,0 +1,114 @@
+    - The intended audience for these standards is both clients and Sui Move programmers.
+    - Clients are aware of the key functions and events in these libraries. They know how to craft transactions using these functions, use read functions for (e.g.) checking ownership of an NFT, and interpret events emitted by the library to inform the user of on-chain happenings.
+    - Sui Move programmers use these libraries both for creating NFT’s (e.g., designing an NFT drop using an auction library) and writing NFT infrastructure that build on top of the core libraries (e.g., creating a new royalty policy that plugs into the extensible libraries)
+
+# Commerce
+
+These standards do not yet exist, but here are:
+
+- the requirements we have in mind
+- the philosophy behind these requirements
+- some implementation ideas
+
+### Requirements
+
+- **Simple proof of ownership for clients:** We must support checking NFT ownership in a uniform way.
+    - This mechanism is intended for off-chain ownership checking: e.g., for a wallet to show a user which NFT’s they own, allowing access to tokengated content, etc. On-chain ownership checking will likely use a separate mechanism that is more direct (e.g., “function `f` checks ownership of a `T` by asking for a parameter of type `&T`).
+    - If all NFT’s were single-owner objects, this would be easy (just look at the object ownership metadata!), but this is too restrictive—many NFT’s will need to be shared or quasi-shared (e.g., an NFT listed for sale on a marketplace will be quasi-shared, but will still have an owner).
+    - This should be simple and broadly accessible: either a single function call, or small number of direct object reads encapsulated behind a single API call. This should *not* require an indexer or special API’s not supported by an ordinary full node
+- **Listing without lockup**. An ****NFT listed for sale (e.g., on a marketplace) must retain most (but not all) of its functionality
+    - NFT’s that are listed for sale should still have an owner, and should be usable for tokengating, games, etc.
+    - However, it is important that listed NFT’s cannot be mutated by the owner (or if it can, this fact + the risks should be very clear to the buyer)—e.g., the owner of a `Hero` listed for sale at a high price should not be able to remove items from the `Hero`'s inventory just before a sale happens.
+    - An NFT should be able to be listed on multiple marketplaces at the same time. Note that the natural way of implementing marketplace listing (make the NFT object a child of a shared object marketplace) does not support this.
+- **Low-level royalty enforcement**. We believe royalties are the raison d'être of NFT’s from the creator perspective—this is a feature physical or web2 digital creations cannot easily have.
```

Comment:
Perhaps a useful angle to divide this problem under is to consider the following scenarios separately:
1. Both client and marketplace _want_ to pay royalties
2. Marketplace wants to avoid royalties but the client does not
3. Client wants to avoid royalties but the marketplace does not
4. Both client and marketplace are disingenuous and want to avoid royalties

Case 4. is fundamentally lost, there will always exist a way for them to settle trades.

Case 1. is an ideal world scenario where we wouldn't have to build anything to perform royalty enforcement.

Case 3. is simple to solve if we let the marketplace tell the `Safe`: "this is how much has been paid for the NFT, now go and do your thing with the appropriate `FooNft::pay_royalty` function." We implemented this today in the OB to see how it would work.

Case 4. is harder to enforce. If most popular wallets want to enforce royalties, what mechanism can we provide them to validate that the marketplace has indeed paid the expected amount, otherwise fail the tx.
If we use the approach of `RoyaltyReceipt` then we must somehow in the marketplace validate that this is indeed the expected amount of royalty paid, otherwise scenario 2. is not covered. Additionally, a client might not always know the exact price they will pay upfront (e.g. contracts where slippage is appropriate such as order book.)